src/components/form/components/select.js

```diff
@@ -36,7 +36,7 @@ const Select = ({
       renderAs="select"
       {...props}
       multiple={multiple}
-      value={value}
+      value={value !== undefined ? value : multiple ? [] : ''}
```

Comment:
nested ternary is unreadable. consider extracting this as a function named for example `getDefaultValue()` or just `defaultValue()`.